spec/wayfinders/wayfinder_spec.rb

```diff
@@ -0,0 +1,703 @@
+
+    describe "#ephemera_vocabulary" do
+      it "returns the vocabulary for an ephemera field" do
+        vocabulary = FactoryBot.create_for_repository(:ephemera_vocabulary)
+        field = FactoryBot.create_for_repository(:ephemera_field, member_of_vocabulary_id: vocabulary.id)
+
+        wayfinder = described_class.for(field)
+
+        expect(wayfinder.ephemera_vocabulary.id).to eq vocabulary.id
+        expect(wayfinder.ephemera_vocabulary.class).to eq EphemeraVocabulary
+      end
+    end
+
+    describe "#decorated_ephemera_vocabulary" do
+      it "returns the vocabulary for an ephemera field decorated" do
+        vocabulary = FactoryBot.create_for_repository(:ephemera_vocabulary)
+        field = FactoryBot.create_for_repository(:ephemera_field, member_of_vocabulary_id: vocabulary.id)
+
+        wayfinder = described_class.for(field)
+
+        expect(wayfinder.decorated_ephemera_vocabulary.id).to eq vocabulary.id
+        expect(wayfinder.decorated_ephemera_vocabulary.class).to eq EphemeraVocabularyDecorator
+      end
+    end
+
+    describe "#ephemera_box" do
+      it "returns the ephemera box for an ephemera folder in a box" do
+        folder = FactoryBot.create_for_repository(:ephemera_folder)
+        box = FactoryBot.create_for_repository(:ephemera_box, member_ids: folder.id)
+
+        wayfinder = described_class.for(folder)
+
+        expect(wayfinder.ephemera_box.id).to eq box.id
+        expect(wayfinder.ephemera_box.class).to eq EphemeraBox
+      end
+    end
+
+    describe "#decorated_ephemera_box" do
+      it "returns the ephemera box for an ephemera folder in a box decorated" do
+        folder = FactoryBot.create_for_repository(:ephemera_folder)
+        box = FactoryBot.create_for_repository(:ephemera_box, member_ids: folder.id)
+
+        wayfinder = described_class.for(folder)
+
+        expect(wayfinder.decorated_ephemera_box.id).to eq box.id
+        expect(wayfinder.decorated_ephemera_box.class).to eq EphemeraBoxDecorator
+      end
+    end
+
+    describe "#ephemera_boxes" do
+      it "returns all box members without accessing members for a project" do
+        box = FactoryBot.create_for_repository(:ephemera_box)
+        folder = FactoryBot.create_for_repository(:ephemera_folder)
+        project = FactoryBot.create_for_repository(:ephemera_project, member_ids: [box.id, folder.id])
+
+        wayfinder = described_class.for(project)
+        allow(wayfinder.query_service).to receive(:find_members).and_call_original
+
+        expect(wayfinder.ephemera_boxes.map(&:id)).to eq [box.id]
+        expect(wayfinder.ephemera_boxes.map(&:class)).to eq [EphemeraBox]
+        expect(wayfinder.query_service).to have_received(:find_members).with(resource: project, model: EphemeraBox)
+
+        expect(wayfinder.decorated_ephemera_boxes.map(&:id)).to eq [box.id]
+        expect(wayfinder.decorated_ephemera_boxes.map(&:class)).to eq [EphemeraBoxDecorator]
+      end
+    end
+
+    describe "#ephemera_folders" do
```

Comment:
We need this and the ones on line 42/55?

Reply:
Yeah - I reorganized, they're for methods that need to exist for different contexts, project vs box. I fixed up the contexts to show that better.